tests/lib/sms/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(sms_test)
+
+# generate runner for the test
+test_runner_generate(src/sms_test.c)
+
+target_include_directories(app PRIVATE src)
+
+cmock_handle(../../../include/modem/at_cmd.h)
+zephyr_compile_definitions(CONFIG_AT_CMD_RESPONSE_MAX_LEN=2700)
```

Comment:
This is a Kconfig setting, why not using Kconfig to control this ?

Reply:
Actually, this can be removed as SMS library is not using CONFIG_AT_CMD_RESPONSE_MAX_LEN anymore. So I've removed this.

CONFIG_AT_CMD_RESPONSE_MAX_LEN in prj.conf of the Unity test didn't help because CONFIG_AT_CMD_RESPONSE_MAX_LEN is not defined by the components included into test run as at_cmd library is mocked. At least this is my impression based on errors about undeclared CONFIG_AT_CMD_RESPONSE_MAX_LEN.

Reply:
but you can do as I write here: https://github.com/nrfconnect/sdk-nrf/pull/3698#discussion_r621117086
> Also, if you just need access to the AT_NOTIF or the AT_NOTIF_SYS_INIT symbols for testing in tests/ then you can provide a test only Kconfig that provides identical symbols that you may need without depends.

Create a `tests/lib/sms/Kconfig` file with the symbols you need to have always available or that needs a different default.
Can also be used to ensure correct values for promptless symbols.

In this particular case you could do:
```
config AT_CMD_RESPONSE_MAX_LEN                                                                      
         int "Maximum AT command response length"                                                    
         default 2700
         help
           Needed to facilitate mocking   

source "Kconfig.zephyr"
```

and the symbol will always be available to your test, regardless of other Kconfig settings.
For more inspiration:
https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/clock_control/nrf_clock_calibration/Kconfig